accounts/config.go

```diff
@@ -0,0 +1,71 @@
+package accounts
+
+import (
+	"errors"
+	"log"
+	"path/filepath"
+
+	"github.com/phomer/scheduler/datastore"
+)
+
+type ClientConfig struct {
+	Hostname    string
+	Port        string
+	Username    string
+	Credentials string
+}
+
+var paths = []string{".", "~/.schedule"}
+
+// Register initialization
+func NewClientConfig(username string, hostname string) *ClientConfig {
+
+	return &ClientConfig{
+		Hostname: hostname,
+		Username: username,
+		Port:     "8000",
+	}
+}
+
+// Client initialization
+func FindClientConfig() *ClientConfig {
+	file := FindFile(paths)
+	buffer := datastore.ReadFile(file)
+
+	config := new(ClientConfig)
+
+	if len(buffer) > 0 {
+		config = datastore.Deserialize(buffer, config).(*ClientConfig)
+	} else {
+		log.Fatal("Missing Config file", errors.New("Missing"), file)
+	}
+
+	return config
+}
+
+func (config *ClientConfig) filename() string {
+	name := config.Hostname + "-" + config.Username + ".key"
+	return name
+}
+
+func (config *ClientConfig) SaveConfig() {
+	if config == nil {
+		log.Fatal("Missing Config Data", nil)
+	}
+
+	buffer := datastore.Serialize(config)
+
+	filename := config.filename()
+	datastore.WriteFile(filepath.Join(".", filename), buffer)
+}
+
+func FindFile(paths []string) string {
+	for i := 0; i < len(paths); i++ {
+		path := "./"
+		filename := "elephant-paulwhomer.key"
+		datastore.TouchFile(path, filename)
+
+		return filepath.Join(path, filename)
+	}
```

Comment:
I fail to see how this actually cycles through the `paths` supplied as the argument

Reply:
Yea, I had started to code it properly, then realized it was a bit messier than expected (I was thinking of calling out to the console if in either dir I hit multiple .key files), then I just hardcoded it to me and my machine to get it to work :-(